app/assets/javascripts/discourse/app/components/sidebar/api-section.gjs

```diff
@@ -27,13 +27,18 @@ export default class SidebarApiSection extends Component {
       return this.section.links;
     }
 
-    if (this.section.text.toLowerCase().match(this.sidebarState.filter)) {
+    if (
+      this.section.text.toLowerCase().match(this.sidebarState.filter.trim())
```

Comment:
Like the lowercasing, we should do the trim directly in the filter component:

https://github.com/discourse/discourse/blob/472c02bda85c2f34da6c4ce90b6e55f84eed64a1/app/assets/javascripts/discourse/app/components/sidebar/filter.gjs#L27-L27